### PR TITLE
Use external packages v2

### PR DESCRIPTION
**Pull request by wolfv**
Checklist
* [ ] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,49 +1,68 @@
...
 
 source:
-    fn: FreeImage{{ version.replace(".", "") }}.zip
-    url: http://downloads.sourceforge.net/freeimage/FreeImage{{ version.replace(".", "") }}.zip
+  - url: http://downloads.sourceforge.net/freeimage/FreeImage{{ version.replace('.', '') }}.zip
     sha256: f41379682f9ada94ea7b34fe86bf9ee00935a3147be41b6569c9605a53e438fd
-    patches:
-        - Makefile.osx.patch  # [osx]
+
+  - path: CMakeLists.txt
```
Comment (xhochy) [inline recipe/meta.yaml:15]: What is the source of these files? Written for this recipe or taken from somewhere else?

Comment (wolfv) [reply recipe/meta.yaml:15]: inspired from VCPKG :)

Comment (wolfv) [reply recipe/meta.yaml:15]: do you want me to add a note?

**Comment by isuruf** [comment]
Have you checked that the current_version and compatbility_version match that of the libraries built by the Makefile?

File: recipe/meta.yaml (hunk 1 above)
Comment (xhochy) [reply recipe/meta.yaml:15]: Would be nice.

**Comment by wolfv** [comment]
@isuruf 

this package is used only by _one_  __old__ version of another package on conda-forge (occt). 
The dependency on freeimage was removed in occt here: https://github.com/conda-forge/occt-feedstock/commit/390507d8bc13425e11649d17f6e6f21d9f489945#diff-e178b687b10a71a3348107ae3154e44c

Given that there is so little exposure, do we need to care?

Otherwise, again, I am happy to de-activate the OS X build for now until I have a mac mini where I can test. 

And another point: for freeimage, ABI version and library version appear to be the same (3.18.0) so maybe this will not be a problem. For Linux I made sure that the SO version is correct.

I have spent wayyyy to much time on this completely useless recipe already so I'd be really glad if we could finally move forward on this :)

**Comment by isuruf** [comment]
> Given that there is so little exposure, do we need to care?

Yes, IMHO. You should ask the maintainers here and clearly state that this build might be incompatible with previous builds. Not mentioning it is wrong.

> I have spent wayyyy to much time on this completely useless recipe already so I'd be really glad if we could finally move forward on this :)

I'm sorry, you keep bringing this up with me, but I'm not paid at all to work on conda-forge and I too spend wayyyy too much time on conda-forge.

**Comment by wolfv** [comment]
so @jakirkham already mentioned to me that he has no time to look at this, and @msarahan told me he's fine with moving to CMake on gitter.
Also, the current build is currently broken when linking to the conda-forge libjpeg as well (on linux) and it's a really bad breakage because it loads two different versions of libjpeg and then segfaults.

So the current state is IMO already broken, that's why I was putting in all this effort -- not to break everything even more. That's why in the beginning I only had a Linux build and you pushed me to make a OS X build which I tried to do using the old school makefiles provided by freeimage (because I was told this is necessary for proper compatibility) and which was really tedious. I care about the quality of conda-forge packages just as much but I don't have a Mac easily available right now to do in-depth testing. We could have just merged with only the Linux build in order to finally fix the broken state of this library which would have allowed me to move forward with gazebo ... 

Let's see what the otool output is and pray that it's the same as from the makefile (0.0.0, and 0.0.0)

**Comment by wolfv** [comment]
Ok so the output from otool indicates that the new libs have:

```
      current version 3.18.0
compatibility version 3.0.0
```

while the old ones had 0.0.0 for both. 

Now I am not sure if that is a wild card or not ... what do you think @isuruf ? should I set SOVERSION to 0 and VERSION 0.0.0 for OS X?

**Comment by isuruf** [comment]
I had a look at OSX package in anaconda.org and the install name is broken. So, it's fine to leave it as it is.

**Comment by wolfv** [comment]
cool, thanks for checking

**Comment by wolfv** [comment]
@xhochy i've added the note.

File: recipe/build.sh [hunk 2]
```diff
@@ -1,19 +1,50 @@
...
+make install
 
-make install PREFIX="${PREFIX}" DESTDIR="${PREFIX}" INCDIR="${PREFIX}/include" INSTALLDIR="${PREFIX}/lib" || exit 1;
+export VERSION=3.18.0
+
+if [ `uname` == Darwin ]; then
+      export SHLIB_EXT=".dylib";
+else
+      export SHLIB_EXT=".so";
+fi
```
Comment (isuruf) [inline recipe/build.sh]: You can remove the 5 lines above

File: recipe/build.sh [hunk 3]
```diff
@@ -1,19 +1,50 @@
...
+      -DCMAKE_FIND_APPBUNDLE=NEVER \
+      ..
 
-mkdir -p ${PREFIX}/lib || exit 1;
-mkdir -p ${PREFIX}/include || exit 1;
+make VERBOSE=1 -j${CPU_COUNT}
+make install
 
-make install PREFIX="${PREFIX}" DESTDIR="${PREFIX}" INCDIR="${PREFIX}/include" INSTALLDIR="${PREFIX}/lib" || exit 1;
+export VERSION=3.18.0
```
Comment (isuruf) [inline recipe/build.sh]:
```suggestion
export VERSION=$PKG_VERSION
```

File: recipe/build.sh [hunk 4]
```diff
@@ -1,19 +1,50 @@
...
+
+if [ `uname` == Darwin ]; then
+      export SHLIB_EXT=".dylib";
+else
+      export SHLIB_EXT=".so";
+fi
+
+cmake -E create_symlink ${PREFIX}/lib/libfreeimage${SHLIB_EXT} ${PREFIX}/lib/libfreeimage-${VERSION}${SHLIB_EXT}
+
+if [ `uname` == Darwin ]; then
```
Comment (isuruf) [inline recipe/build.sh]:
```suggestion
if [[ "$target_platform" == "osx-64" ]]; then
```

File: recipe/build.sh [hunk 5]
```diff
@@ -1,19 +1,50 @@
...
+      otool -l ${PREFIX}/lib/libfreeimage.dylib
```
Comment (isuruf) [inline recipe/build.sh]:
```suggestion
      $OTOOL -l ${PREFIX}/lib/libfreeimage.dylib
```

**Review by isuruf — APPROVED** [review]

**Pull request merged**